server/src/main/java/org/elasticsearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -1411,6 +1411,17 @@ private static void updateIndexMappingsAndBuildSortOrder(
         }
     }
 
+    private static void validateSourceMode(MapperService mapperService) {
+        final IndexSettings indexSettings = mapperService.getIndexSettings();
+        if (indexSettings != null
+            && indexSettings.getIndexVersionCreated().onOrAfter(IndexVersions.DEPRECATE_SOURCE_MODE_MAPPER)
+            && SourceFieldMapper.isSynthetic(indexSettings) == false
```

Comment:
I think we need to check whether `index.mapping.source.mode` index setting has not be defined here? Other we only check whether this setting has been set to `synthetic`?